ngraph/core/src/function.cpp

```diff
@@ -520,3 +516,78 @@ ov::op::util::Variable::Ptr ov::Function::get_variable_by_id(const string& varia
 }
 
 constexpr ov::DiscreteTypeInfo ov::AttributeAdapter<shared_ptr<ov::Function>>::type_info;
+
+/// Output functions
+ov::Output<ov::Node> ov::Function::output(size_t i) const {
+    return m_results.at(i);
+}
+ov::Output<ov::Node> ov::Function::output(const std::string& tensor_name) const {
+    for (const auto& res : m_results) {
+        for (const auto& name : res->get_input_tensor(0).get_names()) {
+            if (name == tensor_name)
+                return res;
+        }
+    }
+    throw ov::Exception("Output for tensor name " + tensor_name + " was not found.");
+}
+const ov::element::Type& ov::Function::output_element_type(size_t i) const {
+    return m_results.at(i)->get_output_element_type(0);
+}
+const ov::element::Type& ov::Function::output_element_type(const std::string& tensor_name) const {
+    for (const auto& res : m_results) {
+        for (const auto& name : res->get_input_tensor(0).get_names()) {
+            if (name == tensor_name)
+                return res->get_output_element_type(0);
+        }
+    }
+    throw ov::Exception("Output for tensor name " + tensor_name + " was not found.");
+}
+const ov::Shape& ov::Function::output_shape(size_t i) const {
+    return m_results.at(i)->get_output_partial_shape(0);
+}
+const ov::Shape& ov::Function::output_shape(const std::string& tensor_name) const {
+    for (const auto& res : m_results) {
+        for (const auto& name : res->get_input_tensor(0).get_names()) {
+            if (name == tensor_name)
+                return res->get_output_partial_shape(0);
```

Comment:
Since a lot of methods use this pattern, can you perhaps add a helper function/method that looks up a result by tensor name and returns a const reference to it? The `output_shape()` methoud would then look like this:
```
const ov::Shape& ov::Function::output_shape(const std::string& tensor_name) const {
    return find_result(tensor_name)->get_output_partial_shape(0);
}
```

Reply:
or just use `output(tensor_name)->get_partial_shape()`